parser_library/src/expressions/mach_operator.h

```diff
@@ -137,7 +142,24 @@ inline mach_expression::value_t mach_expr_binary<sub>::evaluate(mach_evaluate_in
 {
     return left_->evaluate(info) - right_->evaluate(info);
 }
-
+template<>
+inline mach_expression::value_t mach_expr_binary<rel_addr>::evaluate(mach_evaluate_info info) const
+{
+    auto lc = left_->evaluate(info);
+    auto symb = right_->evaluate(info);
+    if (symb.value_kind() == context::symbol_value_kind::ABS)
+    {
+        add_diagnostic(diagnostic_op::warn_D031(get_range(), std::to_string(right_->evaluate(info).get_abs())));
+        return right_->evaluate(info);
+    }
+    else
+    {
+        if (lc.get_reloc().offset() > symb.get_reloc().offset())
+            return (left_->evaluate(info) - right_->evaluate(info));
+        else
+            return (right_->evaluate(info) - left_->evaluate(info));
```

Comment:
I don't understand this part... relative address may very well be positive or negative.
I'd expect simply `target-location`.

Reply:
Okay, thanks for pointing out. I misunderstood it as difference should be always positive value, will change it to target - location.